client/app/pages/dashboards/hooks/useDashboard.js

```diff
@@ -95,7 +96,16 @@ function useDashboard(dashboardData) {
   const loadWidget = useCallback((widget, forceRefresh = false) => {
     widget.getParametersDefs(); // Force widget to read parameters values from URL
     setDashboard(currentDashboard => extend({}, currentDashboard));
-    return widget.load(forceRefresh).finally(() => setDashboard(currentDashboard => extend({}, currentDashboard)));
+    return widget
+      .load(forceRefresh)
+      .catch(error => {
+        // QueryResultErrors are expected
+        if (error instanceof QueryResultError) {
+          return;
+        }
+        return error;
```

Comment:
If this error should be propagated to next `.catch` handler - you should return rejected promise here (e.g. `return Promise.reject(error)`), otherwise the flow will continue with next `.then` handler.

Reply:
Also, if this error is okay for widget - what do you think about moving this `.catch` to a Widget service to the `load` method?

Reply:
> If this error should be propagated to next .catch handler - you should return rejected promise here (e.g. return Promise.reject(error)), otherwise the flow will continue with next .then handler.

Good catch, thanks! 👍

> Also, if this error is okay for widget - what do you think about moving this .catch to a Widget service to the load method?

It is an expected error, but I preferred to keep it propagated, so we can decide what to do from load calls (e.g.: if we eventually decide to add notifications after loading).